samples/cpp/benchmark_app/inputs_filling.cpp

```diff
@@ -549,69 +800,70 @@ std::map<std::string, ov::TensorVector> get_tensors_static_case(const std::vecto
         }
     }
 
-    size_t imageInputsNum = net_input_im_sizes.size();
-    size_t binaryInputsNum = app_inputs_info.size() - imageInputsNum;
+    std::vector<std::string> binaryFiles = filter_files_by_extensions(inputFiles, supported_binary_extensions);
```

Comment:
I had to modify it because the numpy files might or might not fall into the 'is_image' category of the AppInputInfo, thus this count would be off / I would not be able to differentiate between image/numpy or binary/numpy.  Instead, I just filtered the files by extensions beforehand and checked their counts afterward.

The aforementioned `input_info.is_image()` function is actually super problematic, and from my understanding, it might cause a lot of issues, when the input shape and layout is provided, it resembles an image, but binary file is provided as an input, so the binary file would not be read correctly. I will verify whether this is an issue, this might require some more changes in the future

Reply:
I confirm that given a model with image-like input, '-layout' flag set to image-like (so that AppInputInfo.is_image() == true) and -i set to binary file causes `[ WARNING ] File data/z_f32.bin cannot be opened!` (from FormatReader) and `Segmentation fault (core dumped)`